app/models/global_role.rb

```diff
@@ -31,4 +31,15 @@ def self.givable
     super
       .where(type: "GlobalRole")
   end
+
+  def self.standard
+    standard_global_role = where(builtin: BUILTIN_STANDARD_GLOBAL).first
+    if standard_global_role.nil?
+      standard_global_role = create(name: "Standard global role", position: 0) do |role|
+        role.builtin = BUILTIN_STANDARD_GLOBAL
+      end
+      raise "Unable to create the standard global role." if standard_global_role.new_record?
+    end
+    standard_global_role
+  end
```

Comment:
Would this not work too?
```suggestion

  def self.standard
    standard_global_role = find_or_create_by(name: "Standard global role", builtin: BUILTIN_STANDARD_GLOBAL, position: 0)
    raise "Unable to create the standard global role." unless standard_global_role.persisted?
    standard_global_role
  end
```

Reply:
The `find_or_create_by(name: "Standard global role", builtin: BUILTIN_STANDARD_GLOBAL, position: 0)` could be problematic if let's say the name or position differs. I would rather use the `builtin: BUILTIN_STANDARD_GLOBAL` as a finding condition only.

Reply:
Okay sure, just wondering the reason behind it. All looks good 🌴

Reply:
Let's say the role already exists in the database, with a different position. Position will be reassigned at some point, because it is managed by the acts as list gem. If we have the `position:0` in the finder conditions, the existing role wouldn't be found and it would try to create a new Role, which is incorrect.
We have this pattern implemented for the ProjectRoles too.

https://github.com/opf/openproject/blob/fb607a24d32585171d55b1fedffc059af17706f4/app/models/project_role.rb#L37-L46